roadmap/implementers-guide/src/node/disputes/dispute-distribution.md

```diff
@@ -10,48 +20,211 @@
 
 - [`DisputeCoordinatorMessage::ActiveDisputes`][DisputeParticipationMessage]
 - [`DisputeCoordinatorMessage::ImportStatements`][DisputeParticipationMessage]
+- [`DisputeCoordinatorMessage::QueryCandidateVotes`][DisputeParticipationMessage]
 - [`RuntimeApiMessage`][RuntimeApiMessage]
 
-## Functionality
+### Wire format
+
+#### Disputes
+
+Protocol: "/polkadot/dispute/1"
+
+Request:
+
+```rust
+struct DisputeRequest {
+  // Either initiating invalid vote or our own (if we voted invalid).
+  invalid_vote: SignedV2<InvalidVote>,
+  // Some invalid vote (can be from backing/approval) or our own if we voted
+  // valid.
+  valid_vote: SignedV2<ValidVote>,
+}
+
+struct InvalidVote {
+  /// The candidate being disputed.
+  candidate_hash: CandidateHash,
+  /// The voting validator.
+  validator_index: ValidatorIndex,
+  /// The session the candidate appears in.
+  candidate_session: SessionIndex,
+}
```

Comment:
Even though it is unnecessary now, the `InvalidDisputeStatementKind` would be helpful for future-proofing, I think

Reply:
uh sure - I somehow missed that we have that.